test/ar_gateway_middleware_tests.erl

```diff
@@ -0,0 +1,577 @@
+-module(ar_gateway_middleware_tests).
```

Comment:
I added a fixup up adding this module to the list in `ar.erl` to make `make test` pick up these tests.